keras_hub/src/models/yolo_v8/non_max_suppression.py

```diff
@@ -0,0 +1,504 @@
+      scores: a tensor with a shape of [batch_size, num_boxes] representing
+        the scores of boxes.
+      boxes: a tensor with a shape of [batch_size, num_boxes, 4] representing
+        the boxes.
+
+    Returns:
+      sorted_scores: a tensor with a shape of [batch_size, num_boxes]
+        representing the sorted scores.
+      sorted_boxes: a tensor representing the sorted boxes.
+      sorted_scores_indices: a tensor with a shape of [batch_size, num_boxes]
+        representing the index of the scores in a sorted descending order.
+    """
+    sorted_scores_indices = ops.flip(
+        ops.cast(ops.argsort(scores, axis=1), "int32"), axis=1
+    )
+    sorted_scores = ops.take_along_axis(
+        scores,
+        sorted_scores_indices,
+        axis=1,
+    )
+    sorted_boxes = ops.take_along_axis(
+        boxes,
+        ops.expand_dims(sorted_scores_indices, axis=-1),
+        axis=1,
+    )
+
+    return sorted_scores, sorted_boxes, sorted_scores_indices
+
+
+def non_max_suppression(
+    boxes,
+    scores,
+    max_output_size,
+    iou_threshold=0.5,
+    score_threshold=0.0,
+    tile_size=512,
+):
+    # Box format must be yxyx
+    """Non-maximum suppression.
+    Ported from https://github.com/tensorflow/tensorflow/blob/v2.12.0/tensorflow/python/ops/image_ops_impl.py#L5368-L5458
+
+    Args:
+      boxes: a tensor with a shape of [batch_size, num_boxes, 4].
+        Dimensions except the last two are batch dimensions. The last dimension
+        represents box coordinates in yxyx format.
+      scores: a tensor with a shape of [batch_size, num_boxes].
+      max_output_size: a scalar integer tensor representing the maximum number
+        of boxes to be selected by non max suppression.
+      iou_threshold: a float representing the threshold for deciding whether boxes
+        overlap too much with respect to IoU (intersection over union).
+      score_threshold: a float representing the threshold for box scores. Boxes
+        with a score that is not larger than this threshold will be suppressed.
+      tile_size: an integer representing the number of boxes in a tile, i.e.,
+        the maximum number of boxes per image that can be used to suppress other
+        boxes in parallel; larger tile_size means larger parallelism and
+        potentially more redundant work.
+
+    Returns:
+      selected_box_args: a tensor with a shape of [..., num_boxes] representing the
+        indices selected by non-max suppression. The leading dimensions
+        are the batch dimensions of the input boxes. All numbers are within
+        [0, num_boxes). For each image (i.e., selected_box_args[i]), only the first
+        num_valid[i] indices (i.e., selected_box_args[i][:num_valid[i]]) are valid.
+      num_valid: a tensor of rank 0 or higher with a shape of [...]
+        representing the number of valid indices in selected_box_args.
+        Its dimensions are the batch dimensions of the input boxes.
+    """  # noqa: E501
```

Comment:
remove these `noqa: E501` everywhere, I don't think we should need them

Reply:
All `noqa` comments within the YOLOv8 model have been removed.